internal/util/validation/formats.go

```diff
@@ -56,6 +65,11 @@ func ValidateOciImageReferenceStrict(s *string, path string) []error {
 	return ValidateString(s, path, 1, OciImageReferenceMaxLength, StrictOciImageReferenceRegexp, StrictOciImageReferenceFmt, "quay.io/flightctl/flightctl:latest")
 }
 
+// Validates an OCI image reference that can contain template parameters.
+func ValidateOciImageReferenceWithTemplates(s *string, path string) []error {
```

Comment:
Looks good, could we add some unit test coverage for this function to codefy the input validation? Thanks!